src/components/dialogs/line-split-with-voltage-level-dialog.js

```diff
@@ -548,11 +539,22 @@ const LineSplitWithVoltageLevelDialog = ({
 
     const lineSubstation = (isFirst) => {
         if (!lineToDivide) return '';
-        const vlId = isFirst
-            ? lineToDivide.voltageLevelId1
-            : lineToDivide.voltageLevelId2;
-        const mayVl = allVoltageLevelOptions.filter((vl) => vl.id === vlId);
-        if (mayVl && mayVl.length) return mayVl[0].name;
+        let vlId = '';
+        if (typeof lineToDivide === 'object') {
+            vlId = isFirst
+                ? lineToDivide.voltageLevelId1
+                : lineToDivide.voltageLevelId2;
+        } else if (isFirst) {
+            const mayLine = lineOptions.filter((l) => l?.id === newLine1Id);
```

Comment:
Maybe use find instead of filter in all this function to avoid access to array[0], a find by id which is unique by the way.

Reply:
done